sdks/java/extensions/zetasketch/src/test/java/org/apache/beam/sdk/extensions/zetasketch/BigQueryHllSketchCompatibilityIT.java

```diff
@@ -0,0 +1,145 @@
+    String query =
+        String.format(
+            "SELECT HLL_COUNT.INIT(%s) AS %s FROM %s",
+            DATA_FIELD_NAME, QUERY_RESULT_FIELD_NAME, tableSpec);
+    SerializableFunction<SchemaAndRecord, byte[]> parseQueryResultToByteArray =
+        (SchemaAndRecord schemaAndRecord) ->
+            // BigQuery BYTES type corresponds to Java java.nio.ByteBuffer type
+            ((ByteBuffer) schemaAndRecord.getRecord().get(QUERY_RESULT_FIELD_NAME)).array();
+
+    TestPipelineOptions options =
+        TestPipeline.testingPipelineOptions().as(TestPipelineOptions.class);
+
+    Pipeline p = Pipeline.create(options);
+    PCollection<Long> result =
+        p.apply(
+                BigQueryIO.read(parseQueryResultToByteArray)
+                    .fromQuery(query)
+                    .usingStandardSql()
+                    .withMethod(Method.DIRECT_READ)
+                    .withCoder(ByteArrayCoder.of()))
+            .apply(HllCount.MergePartial.globally()) // no-op, only for testing MergePartial
+            .apply(HllCount.Extract.globally());
+    PAssert.thatSingleton(result).isEqualTo(EXPECTED_COUNT);
+    p.run().waitUntilFinish();
+  }
+
+  /**
+   * Test that HLL++ sketch computed in Beam can be processed by BigQuery. Hll sketch is computed by
+   * {@link HllCount.Init} in Beam and written to BigQuery; the test verifies that we can run {@code
+   * HLL_COUNT.EXTRACT()} on the sketch in BigQuery to get the correct estimated count.
+   */
+  @Test
+  public void testWriteSketchToBigQuery() {
+    String tableSpec = String.format("%s.%s", DATASET_NAME, SKETCH_TABLE_NAME);
+    String query =
+        String.format("SELECT HLL_COUNT.EXTRACT(%s) FROM %s", SKETCH_FIELD_NAME, tableSpec);
+    TableSchema tableSchema =
+        new TableSchema()
+            .setFields(
+                Collections.singletonList(
+                    new TableFieldSchema().setName(SKETCH_FIELD_NAME).setType("BYTES")));
+
+    TestPipelineOptions options =
+        TestPipeline.testingPipelineOptions().as(TestPipelineOptions.class);
+    // After the pipeline finishes, BigqueryMatcher will send a query to retrieve the estimated
+    // count and verifies its correctness using checksum.
+    options.setOnSuccessMatcher(
+        BigqueryMatcher.createUsingStandardSql(
+            options.as(ApplicationNameOptions.class).getAppName(),
+            options.as(GcpOptions.class).getProject(),
+            query,
+            EXPECTED_CHECKSUM));
+
+    Pipeline p = Pipeline.create(options);
+    p.apply(Create.of(TEST_DATA))
+        .apply(HllCount.Init.forStrings().globally())
+        .apply(
+            BigQueryIO.<byte[]>write()
+                .to(tableSpec)
+                .withSchema(tableSchema)
+                .withFormatFunction(sketch -> new TableRow().set(SKETCH_FIELD_NAME, sketch))
+                .withWriteDisposition(BigQueryIO.Write.WriteDisposition.WRITE_TRUNCATE));
```

Comment:
Same above. Without creating dataset and table, how can you write to a table?

Reply:
Same above. The dataset is already created, and for write test actually we don't need to create the table. I will be created (or overridden) by the pipeline each time we run the test. Specified here:
https://github.com/apache/beam/blob/c5ddcf6695614dbeb136a36c8750c3b458ee26bb/sdks/java/extensions/zetasketch/src/test/java/org/apache/beam/sdk/extensions/zetasketch/BigQueryHllSketchCompatibilityIT.java#L64